packages/engine/Source/Scene/Scene.js

```diff
@@ -2410,31 +2395,168 @@ function executeCommands(scene, passState) {
         );
       };
     }
-    executeTranslucentCommands = scene._executeOITFunction;
-  } else if (passes.render) {
-    executeTranslucentCommands = executeTranslucentCommandsBackToFront;
-  } else {
-    executeTranslucentCommands = executeTranslucentCommandsFrontToBack;
+    return scene._executeOITFunction;
+  }
+  if (scene.frameState.passes.render) {
+    return executeTranslucentCommandsBackToFront;
+  }
+  return executeTranslucentCommandsFrontToBack;
+}
+
+/**
+ * Execute draw commands to render translucent objects in the scene.
+ *
+ * @param {Scene} scene The scene.
+ * @param {PassState} passState The state for the current render pass.
+ * @param {FrustumCommands} frustumCommands The draw commands for the current frustum.
+ *
+ * @private
+ */
+function performTranslucentPass(scene, passState, frustumCommands) {
+  const { frameState, context } = scene;
+  const { pick, pickVoxel } = frameState.passes;
+  const picking = pick || pickVoxel;
+
+  let invertClassification;
+  if (
+    !picking &&
+    scene._environmentState.useInvertClassification &&
+    frameState.invertClassificationColor.alpha < 1.0
+  ) {
+    // Fullscreen pass to copy unclassified fragments when alpha < 1.0.
+    // Not executed when undefined.
+    invertClassification = scene._invertClassification;
+  }
+
+  const executeTranslucentCommands = obtainTranslucentCommandExecutionFunction(
+    scene
+  );
+
+  context.uniformState.updatePass(Pass.TRANSLUCENT);
+  const commands = frustumCommands.commands[Pass.TRANSLUCENT];
+  commands.length = frustumCommands.indices[Pass.TRANSLUCENT];
```

Comment:
I know that this was **not changed** in this PR (i.e. it has already been like that before the changes here). But I recently asked about that line, and for me, this _strongly_ looks like an ancient copy-and-paste error: Usually, the `length` is fetched from this `indices` array. But here, the length is **assigned to** the array of commands. Are we sure that this is correct and makes sense?

Reply:
> Are we sure that this is correct and makes sense?
I'm not sure about either of those, but I would say that it is currently necessary.

The structure of `FrustumCommands`, as modified by the `insertIntoBin` function in `View`, does not (obviously) imply that we can depend on the `commands` array to have a useful `.length` property. However, the `executeTranslucentCommands` function eventually may pass the `commands` array to `mergeSort`, which _does_ assume that the array should be sorted along its `length`. Alternatively, it will be sent to `OIT.prototype.executeCommands`, which also uses the length.

I didn't want to change `Core/mergeSort` in this PR, so for now we do need to set the `.length` property. `performVoxelsPass` has the same issue.